pkgs/build-support/dotnet/build-dotnet-module/default.nix

```
@@ -134,45 +150,87 @@ in stdenvNoCC.mkDerivation (args // {
   passthru = {
     inherit nuget-source;
 
-    fetch-deps = let
-      exclusions = dotnet-sdk.passthru.packages { fetchNuGet = attrs: attrs.pname; };
-    in writeShellScript "fetch-${pname}-deps" ''
-      set -euo pipefail
-      export PATH="${lib.makeBinPath [ coreutils dotnet-sdk nuget-to-nix ]}"
-
-      cd "$(dirname "''${BASH_SOURCE[0]}")"
-
-      export HOME=$(mktemp -d)
-      deps_file="''${1:-/tmp/${pname}-deps.nix}"
-
-      store_src="${srcOnly args}"
-      src="$(mktemp -d /tmp/${pname}.XXX)"
-      cp -rT "$store_src" "$src"
-      chmod -R +w "$src"
-
-      trap "rm -rf $src $HOME" EXIT
-      pushd "$src"
-
-      export DOTNET_NOLOGO=1
-      export DOTNET_CLI_TELEMETRY_OPTOUT=1
-
-      mkdir -p "$HOME/nuget_pkgs"
-
-      for project in "${lib.concatStringsSep "\" \"" ((lib.toList projectFile) ++ lib.optionals (testProjectFile != "") (lib.toList testProjectFile))}"; do
-        dotnet restore "$project" \
-          ${lib.optionalString (!enableParallelBuilding) "--disable-parallel"} \
-          -p:ContinuousIntegrationBuild=true \
-          -p:Deterministic=true \
-          --packages "$HOME/nuget_pkgs" \
-          ${lib.optionalString (dotnetRestoreFlags != []) (builtins.toString dotnetRestoreFlags)} \
-          ${lib.optionalString (dotnetFlags != []) (builtins.toString dotnetFlags)}
-      done
-
-      echo "${lib.concatStringsSep "\n" exclusions}" > "$HOME/package_exclusions"
-
-      echo "Writing lockfile..."
-      nuget-to-nix "$HOME/nuget_pkgs" "$HOME/package_exclusions" > "$deps_file"
-      echo "Succesfully wrote lockfile to: $deps_file"
-    '';
-  } // args.passthru or {};
+    fetch-deps =
+      let
+        # Because this list is rather long its put in its own store path to maintain readability of the generated script
+        exclusions = writeText "nuget-package-exclusions" (lib.concatStringsSep "\n" (dotnet-sdk.passthru.packages { fetchNuGet = attrs: attrs.pname; }));
+
+        # Derivations may set flags such as `--runtime <rid>` based on the host platform to avoid restoring/building nuget dependencies they dont have or dont need.
+        # This introduces an issue; In this script we loop over all platforms from `meta` and add the RID flag for it, as to fetch all required dependencies.
+        # The script would inherit the RID flag from the derivation based on the platform building the script, and set the flag for any iteration we do over the RIDs.
+        # That causes conflicts. To circumvent it we remove all occurances of the flag.
+        flags =
+          let
+            hasRid = flag: builtins.any (v: v) (map (rid: lib.hasInfix rid flag) (lib.attrValues dotnet-sdk.runtimeIdentifierMap));
```

```suggestion
            hasRid = flag: lib.any (v: v) (map (rid: lib.hasInfix rid flag) (lib.attrValues dotnet-sdk.runtimeIdentifierMap));
```

https://github.com/NixOS/nixpkgs/blob/master/lib/lists.nix#L215